File: PRNumberBiz/app/src/main/java/com/pplus/prnumberbiz/apps/setting/ui/SecessionPreCautionFragment.java
package com.pplus.prnumberbiz.apps.setting.ui;


import android.app.Activity;
import android.content.Intent;
import android.text.Html;
import android.view.View;
import android.widget.TextView;

import com.pple.pplus.utils.part.format.FormatUtil;
import com.pplus.prnumberbiz.Const;
import com.pplus.prnumberbiz.R;
import com.pplus.prnumberbiz.apps.common.builder.AlertBuilder;
import com.pplus.prnumberbiz.apps.common.builder.OnAlertResultListener;
import com.pplus.prnumberbiz.apps.common.builder.data.AlertData;
import com.pplus.prnumberbiz.apps.common.mgmt.LoginInfoManager;
import com.pplus.prnumberbiz.apps.common.ui.base.BaseFragment;
import com.pplus.prnumberbiz.apps.signup.ui.VerificationMeActivity;
import com.pplus.prnumberbiz.core.network.ApiBuilder;
import com.pplus.prnumberbiz.core.network.model.dto.Verification;
import com.pplus.prnumberbiz.core.network.model.response.NewResultResponse;
import com.pplus.prnumberbiz.core.util.PplusCommonUtil;

import java.util.HashMap;
import java.util.Map;

import network.common.PplusCallback;
import retrofit2.Call;


/**
 * 회원 탈퇴 First - 진행방법 및 대기기간
 */
public class SecessionPreCautionFragment extends BaseFragment<SecessionActivity>{


    @Override
    public String getPID(){

        return "";
    }

    @Override
    public int getLayoutResourceId(){

        return R.layout.fragment_secession_pre_caution;
    }

    @Override
    public void initializeView(View container){

        ((TextView) container.findViewById(R.id.tv_secession_caution)).setText(PplusCommonUtil.Companion.fromHtml(getResources().getString(R.string.html_msg_leave_caution, FormatUtil.getMoneyType(""+LoginInfoManager.getInstance().getUser().getTotalCash()), FormatUtil.getMoneyType(""+LoginInfoManager.getInstance().getUser().getTotalBol()))));
        ((TextView) container.findViewById(R.id.tv_secession_pre_caution)).setText(PplusCommonUtil.Companion.fromHtml(getResources().getString(R.string.html_msg_leave_pre_caution)));

        container.findViewById(R.id.text_secession).setOnClickListener(this);
    }

    public SecessionPreCautionFragment(){
        // Required empty public constructor
    }

    @Override
    public void onClick(View v){

        switch (v.getId()) {
            case R.id.text_secession:

                if(PplusCommonUtil.Companion.getCountryCode().equalsIgnoreCase("kr")) {
                    AlertBuilder.Builder builder = new AlertBuilder.Builder();
                    builder.setTitle(getString(R.string.word_notice_alert));
                    builder.addContents(new AlertData.MessageData(getString(R.string.msg_verification_me_alert_secession), AlertBuilder.MESSAGE_TYPE.TEXT, 2));
                    builder.addContents(new AlertData.MessageData(getString(R.string.msg_move_verification), AlertBuilder.MESSAGE_TYPE.TEXT, 2));
                    builder.setLeftText(getString(R.string.word_cancel)).setRightText(getString(R.string.word_confirm));
                    builder.setOnAlertResultListener(new OnAlertResultListener(){

                        @Override
                        public void onCancel(){

                        }

                        @Override
                        public void onResult(AlertBuilder.EVENT_ALERT event_alert){

                            switch (event_alert) {
                                case RIGHT:
                                    Intent intent = new Intent(getActivity(), VerificationMeActivity.class);
                                    intent.putExtra(Const.KEY, Const.VERIFICATION);
                                    intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP);
                                    getActivity().startActivityForResult(intent, Const.REQ_LEAVE);
                                    break;
                            }
                        }
                    }).builder().show(getActivity());

                } else {
                    getParentActivity().secessionAuth();
                }

                break;
        }
    }

    @Override
    public void init(){

    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data){

        super.onActivityResult(requestCode, resultCode, data);
        if(resultCode == Activity.RESULT_OK) {
            switch (requestCode) {
                case Const.REQ_LEAVE:
                    if(data != null) {
                        Verification verification = data.getParcelableExtra(Const.VERIFICATION);
                        if(verification == null) {
                            return;
                        }
                        Map<String, String> params = new HashMap<>();
                        params.put("number", verification.getNumber());
                        params.put("token", verification.getToken());
                        showProgress("");
                        ApiBuilder.create().leave(params).setCallback(new PplusCallback<NewResultResponse<Object>>(){

                            @Override
                            public void onResponse(Call<NewResultResponse<Object>> call, NewResultResponse<Object> response){

                                hideProgress();
                                getParentActivity().setLeave(true);
                                getParentActivity().secessionResult();
                            }

                            @Override
                            public void onFailure(Call<NewResultResponse<Object>> call, Throwable t, NewResultResponse<Object> response){

                                hideProgress();
                            }
                        }).build().call();
                    }

                    break;
            }
        }

    }
}

File: PRNumber/app/src/main/java/com/pplus/prnumberuser/apps/setting/ui/SecessionPreCautionFragment.kt
package com.pplus.prnumberuser.apps.setting.ui

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.activity.result.contract.ActivityResultContracts
import com.pplus.networks.common.PplusCallback
import com.pplus.prnumberuser.Const
import com.pplus.prnumberuser.R
import com.pplus.prnumberuser.apps.common.mgmt.LoginInfoManager
import com.pplus.prnumberuser.apps.common.ui.base.BaseFragment
import com.pplus.prnumberuser.apps.signup.ui.VerificationMeActivity
import com.pplus.prnumberuser.core.network.ApiBuilder
import com.pplus.prnumberuser.core.network.model.dto.Verification
import com.pplus.prnumberuser.core.network.model.response.NewResultResponse
import com.pplus.prnumberuser.core.util.PplusCommonUtil
import com.pplus.prnumberuser.core.util.PplusCommonUtil.Companion.getCountryCode
import com.pplus.prnumberuser.databinding.FragmentSecessionPreCautionBinding
import com.pplus.utils.part.format.FormatUtil
import retrofit2.Call
import java.util.*

/**
 * 회원 탈퇴 First - 진행방법 및 대기기간
 */
class SecessionPreCautionFragment : BaseFragment<SecessionActivity>() {
    override fun getPID(): String? {
        return ""
    }

    private var _binding: FragmentSecessionPreCautionBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {

        _binding = FragmentSecessionPreCautionBinding.inflate(inflater, container, false)
        val view = binding.root
        return view
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    override fun init() {

        binding.tvSecessionCaution.text = PplusCommonUtil.fromHtml(getString(R.string.html_msg_leave_caution))

        //        ((TextView) container.findViewById(R.id.text_secession_retention_cash)).setText(getString(R.string.format_money_unit, FormatUtil.getMoneyType("" + LoginInfoManager.getInstance().getUser().getTotalCash())));
        binding.textSecessionRetentionPoint.text = getString(R.string.format_cash_unit, FormatUtil.getMoneyType("" + LoginInfoManager.getInstance().user.totalBol))


        if (LoginInfoManager.getInstance().user.page != null) {
            binding.textSecessionStep1.setText(R.string.msg_user_withdrawal_step1_biz)
            binding.textSecessionStep2.setText(R.string.msg_user_withdrawal_step2_biz)
        } else {
            binding.textSecessionStep1.setText(R.string.msg_user_withdrawal_step1_user)
            binding.textSecessionStep2.setText(R.string.msg_user_withdrawal_step2_user)
        }
        binding.textSecession.setOnClickListener {
            if (LoginInfoManager.getInstance().user.verification!!.media == "external") {
                val intent = Intent(activity, VerificationMeActivity::class.java)
                intent.putExtra(Const.KEY, Const.VERIFICATION)
                intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_SINGLE_TOP
                leaveLauncher.launch(intent)
            } else {
                val intent = Intent(activity, VerificationActivity::class.java)
                intent.putExtra(Const.KEY, Const.LEAVE)
                intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_SINGLE_TOP
                leaveLauncher.launch(intent)
            }
        }
    }

    val leaveLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
        if(result.resultCode == Activity.RESULT_OK){
            if (result.data != null) {
                val data = result.data!!
                val verification = data.getParcelableExtra<Verification>(Const.VERIFICATION)
                val params = HashMap<String, String>()
                params["number"] = verification!!.number
                params["token"] = verification.token
                params["mobile"] = data.getStringExtra(Const.MOBILE_NUMBER)!!
                params["appType"] = Const.APP_TYPE
                showProgress("")
                ApiBuilder.create().leave(params).setCallback(object : PplusCallback<NewResultResponse<Any>> {
                    override fun onResponse(call: Call<NewResultResponse<Any>>?, response: NewResultResponse<Any>?) {
                        hideProgress()
                        getParentActivity().secessionResult()
                    }

                    override fun onFailure(call: Call<NewResultResponse<Any>>?, t: Throwable?, response: NewResultResponse<Any>?) {
                        hideProgress()
                    }
                }).build().call()
            }
        }
    }

    companion object {

        @JvmStatic
        fun newInstance() = SecessionPreCautionFragment().apply {
            arguments = Bundle().apply {
            }
        }
    }
}